WBooks/Views/AddNew/AddNewViewController.swift

```diff
@@ -8,20 +8,108 @@
         super.viewDidLoad()
         
         _view.configureView()
         
+        let tap = UITapGestureRecognizer(target: self, action: #selector(self.handleTap(_:)))
+        
+        _view.imageButton.addGestureRecognizer(tap)
+        
+        _view.imageButton.isUserInteractionEnabled = true
+        
+        _view.submitButton.addTarget(self, action: #selector(onTap(_:)), for: .touchUpInside)
+        
+    }
+    
+    @objc func handleTap(_ sender: UITapGestureRecognizer) {
+        let alert = UIAlertController(title: "select source", message: "",
+                                      preferredStyle: UIAlertControllerStyle.alert)
+        alert.addAction(UIAlertAction(title: "Gallery", style: .default, handler: { action in
+            switch action.style {
+            case .default:
+                if UIImagePickerController.isSourceTypeAvailable(.photoLibrary) {
+                    let imagePicker = UIImagePickerController()
+                    imagePicker.delegate = self
+                    imagePicker.sourceType = .photoLibrary
+                    imagePicker.allowsEditing = true
+                    self.present(imagePicker, animated: true, completion: nil)
+                }
+                
+            case .cancel:
+                break
+                
+            case .destructive:
+                break
+            }
+        }))
+        alert.addAction(UIAlertAction(title: "Camera", style: .default, handler: { action in
+            switch action.style {
+            case .default:
+                if UIImagePickerController.isSourceTypeAvailable(.camera) {
+                    let imagePicker = UIImagePickerController()
+                    imagePicker.delegate = self
+                    imagePicker.sourceType = .camera
+                    imagePicker.allowsEditing = false
+                    self.present(imagePicker, animated: true, completion: nil)
+                }
+            case .cancel:
+                break
+                
+            case .destructive:
+                break
+            }
+        }))
+        self.present(alert, animated: true, completion: nil)
+//        if UIImagePickerController.isSourceTypeAvailable(.photoLibrary) {
+//            var imagePicker = UIImagePickerController()
+//            imagePicker.delegate = self
+//            imagePicker.sourceType = .photoLibrary;
+//            imagePicker.allowsEditing = true
+//            self.present(imagePicker, animated: true, completion: nil)
+//        }
+        
+    }
+    @objc func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String: AnyObject]) {
+        //swiftlint:disable:next force_cast
+        let image = info[UIImagePickerControllerOriginalImage] as! UIImage
+        _view.bookImage.image = image
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc func onTap(_ sender: Any) {
+        let image: UIImage = _view.bookImage.image!
```

Comment:
Why are you doing force casting? If the user does not set a property this will crash! Also if there are required properties users should not be able to tap on the button.